common/diagnostic.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+//
+// Diagnostics control macros and options
+//
+
+
+//
+// Enable asserts which can terminate the program when exceptional conditions are encountered
+//
+#if defined(AT_DIAG_ENABLE_ASSERT)
+    #include <assert.h>
+    #define AT_ASSERT(expr) assert(expr)
+    #define AT_ASSERT_MSG(expr, format, ... ) if (!(expr)) { fprintf(stderr, format , ##__VA_ARGS__); assert(0); }
```

Comment:
@christian-rauch here's the implementation of AT_ASSERT - it is just defined to assert(expr) in the case where it's enabled or empty otherwise.

Reply:
So `AT_ASSERT` is just implemented by `assert`. But is there a benefit of enabeling/disabeling this manually? Assertions will be disabled in Release mode anyway and having them enabled in Debug mode makes a lot of sense.